### PR TITLE
Make release script runnable locally

### LINKED ISSUE #1310: Pilot Release script failing
More info at https://console.cloud.google.com/gcr/builds/df625fc8-bc62-4e9f-af05-721885f7c971?project=istio-io&organizationId=433637338589

+ source /tmp/gopath/src/istio.io/pilot/bin/use_bazel_go.sh
++++ dirname /tmp/gopath/src/istio.io/pilot/bin/use_bazel_go.sh
+++ cd /tmp/gopath/src/istio.io/pilot/bin
+++ pwd -P
*** /workspace/bin/../bazel-pilot does not exist - did you forget to bazel build ... ?
++ SP=/workspace/bin
++ TARGETBIN=/workspace/bin/../bazel-pilot
++ [[ ! -e /workspace/bin/../bazel-pilot ]]
++ echo '*** /workspace/bin/../bazel-pilot does not exist - did you forget to bazel build ... ?'
++ exit 1

### PR DESCRIPTION
**Pull request by sebastienvas**
Fixes issue in use_bazel_go.sh for release.
Fixes #1310 

Changed the script to be a little more robust such that we can test it locally. Cloud Builder checks out code in /workspace, use_bazel_go assumes the codes is checked out in a pilot directory which is not true. Even so we run the build from the GOPATH, bazel will use the link target to create a bazel-workspace directory.

```release-note
none
```

### CODE REVIEW

**Comment by sebastienvas** [comment]
Tested at https://console.cloud.google.com/gcr/builds/ca992c62-02e7-4e23-92da-5a72c87f9ed8?project=istio-io&authuser=0

File: bin/release.sh [hunk 1]
```diff
@@ -28,10 +28,17 @@ if [ ! "${TAG_NAME}" ] ; then
 fi
 
 # Build environment setup
-mkdir -p /tmp/gopath/src/istio.io
-ln -s /workspace /tmp/gopath/src/istio.io/pilot
-cd /tmp/gopath/src/istio.io/pilot
-touch platform/kube/config
+function prepare_gopath() {
```
Comment (xiaolanz) [inline bin/release.sh]: please add comment to help understand what it does.

File: bin/use_bazel_go.sh [hunk 2]
```diff
@@ -1,9 +1,10 @@
 # This file should be sourced before using go commands
 # it ensures that bazel's version of go is used
 
-SP=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+ROOT=$( cd "$(dirname "${BASH_SOURCE[0]}")/.." ; pwd -P )
+DIR_NAME="$(basename ${ROOT})"
 
-TARGETBIN=$SP/../bazel-pilot
+TARGETBIN="${ROOT}/bazel-${DIR_NAME}"
```
Comment (xiaolanz) [inline bin/use_bazel_go.sh]: why  dir can be other names?

**Comment by xiaolanz** [comment]
would you please add more description in the PR request on what causes the problem and why the fix will resolve it?

File: bin/use_bazel_go.sh (hunk 2 above)
Comment (kyessenov) [reply bin/use_bazel_go.sh]: Please use `bazel info bazel-bin`

Comment (sebastienvas) [reply bin/use_bazel_go.sh]: cloud builder checks out the code into /tmp/workspace and therefore the name is bazel-workspace

Comment (ZackButcher) [reply bin/use_bazel_go.sh]: Oh cool, didn't know `bazel info` did that. Yeah it's definitely better to use here: it gives you the full path to where the files that are symlinked into bazel-bin actually live on the file system. Should be more reliable than this.

**Comment by sebastienvas** [comment]
@xiaolanz  PTAL

File: bin/release.sh (hunk 1 above)
Comment (sebastienvas) [reply bin/release.sh]: Done

File: bin/use_bazel_go.sh (hunk 2 above)
Comment (sebastienvas) [reply bin/use_bazel_go.sh]: @kyessenov that s not bazel-bin that s bazel-pilot

**Pull request merged**